pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -106,22 +119,24 @@ public static void updateOrderStatus(BaseMercuryStatusMessage baseMercuryStatusM
                             null, statusDate, null,
                             rs.getLong(DBConstants.TISSUE_ID), rs.getLong(DBConstants.DSM_KIT_REQUEST_ID), null));
                 } else {
-                    dbVals.resultException = new RuntimeException(
-                            "Couldn't get ddp instance id for order " + mercuryStatusMessage.getOrderID());
+                    dbVals.resultException = new DsmInternalError(String.format("Couldn't find an order based on the order id %s",
+                            mercuryStatusMessage.getOrderID()));
                 }
             } catch (SQLException ex) {
                 dbVals.resultException =
-                        new RuntimeException("Error getting ddp instance id for order " + mercuryStatusMessage.getOrderID(), ex);
+                        new DsmInternalError(String.format(
+                                "Couldn't find an order based on the order id %s, ES update will fail. Status message received was %s",
+                                mercuryStatusMessage.getOrderID(), baseMercuryStatusMessage), ex);
             }
             return dbVals;
         });
         if (results.resultException != null) {
-            throw new DSMPubSubException("Unable to process the status of the order " + baseMercuryStatusMessage, results.resultException);
+            throw (DsmInternalError) results.resultException;
```

Comment:
I assume this cast is unnecessary? Did something suggest you add it?

Reply:
Yes otherwise we have to add `@sneakythrows` or throws Exception to method signature